packages/integration-tests/index.js

```diff
@@ -656,18 +655,30 @@ async function buildWorkflow(
   const template = await getJsonS3Object(bucketName, templateKey(stackName));
 
   if (collection) {
-    template.meta.collection = await collectionsApi.getCollection({
+    const collectionsApiResponse = await collectionsApi.getCollection({
       prefix: stackName,
       collectionName: collection.name,
       collectionVersion: collection.version,
     });
+    if (collectionsApiResponse.statusCode) {
+      throw new Error(`Collections API responded with error on buildWorkflow ${JSON.stringify(collectionsApiResponse)}`);
+    }
+    template.meta.collection = collectionsApiResponse;
```

Comment:
There's an argument to be made here for an API refactor to *not* parse the JSON/return like this, but that's a separate PR/issue to be decided.